src/wallet/wallet2.cpp

```diff
@@ -4983,129 +4948,126 @@ void wallet2::restore(const std::string& wallet_, const epee::wipeable_string& p
-    }
-    else
-    {
-      // M / N case
-      MINFO("Preparing keys for next exchange round...");
+  cryptonote::multisig_kex_account_update(expanded_msgs, multisig_account);
 
-      // Preparing data for middle round - packing new public multisig keys to exchage with others.
-      extra_multisig_info = pack_multisignature_keys(MULTISIG_EXTRA_INFO_MAGIC, derivations, m_account.get_keys().m_spend_secret_key);
-      spend_skey = rct::sk2rct(m_account.get_keys().m_spend_secret_key);
+  CHECK_AND_ASSERT_THROW_MES(multisig_account.m_kex_rounds_complete == 1,
+    "'Rounds complete' has unexpected value while making multisig.");
 
-      // Need to store middle keys to be able to proceed in case of wallet shutdown.
-      m_multisig_derivations = derivations;
-    }
-  }
-  
+  // update wallet state
   if (!m_original_keys_available)
   {
     // Save the original i.e. non-multisig keys so the MMS can continue to use them to encrypt and decrypt messages
     // (making a wallet multisig overwrites those keys, see account_base::make_multisig)
-    m_original_address = m_account.get_keys().m_account_address;
-    m_original_view_secret_key = m_account.get_keys().m_view_secret_key;
+    m_original_address = get_account().get_keys().m_account_address;
+    m_original_view_secret_key = get_account().get_keys().m_view_secret_key;
     m_original_keys_available = true;
   }
 
   clear();
-  MINFO("Creating view key...");
-  crypto::secret_key view_skey = cryptonote::generate_multisig_view_secret_key(get_account().get_keys().m_view_secret_key, view_keys);
 
+  // account base
   MINFO("Creating multisig address...");
-  CHECK_AND_ASSERT_THROW_MES(m_account.make_multisig(view_skey, rct::rct2sk(spend_skey), rct::rct2pk(spend_pkey), multisig_keys),
-      "Failed to create multisig wallet due to bad keys");
-  memwipe(&spend_skey, sizeof(rct::key));
+  CHECK_AND_ASSERT_THROW_MES(m_account.make_multisig(multisig_account.m_aggregate_ancillary_privkey,
+    blinded_skey,
+    multisig_account.m_aggregate_pubkey,
+    multisig_account.m_privkeys),
+      "Failed to create multisig wallet account due to bad keys");
 
   init_type(hw::device::device_type::SOFTWARE);
   m_original_keys_available = true;
   m_multisig = true;
   m_multisig_threshold = threshold;
-  m_multisig_signers = multisig_signers;
-  ++m_multisig_rounds_passed;
+  m_multisig_signers = signers;
+  m_multisig_rounds_passed = 1;
+
+  // derivations stored (should be empty in last round)
+  // TODO: make use of the origins map for aggregation-style signing (instead of round-robin)
+  m_multisig_derivations.clear();
+  m_multisig_derivations.reserve(multisig_account.m_kex_keys_to_origins_map.size());
+
+  for (const auto &key_to_origins : multisig_account.m_kex_keys_to_origins_map)
+    m_multisig_derivations.push_back(key_to_origins.first);
+
+  // address
+  m_account_public_address.m_spend_public_key = multisig_account.m_aggregate_pubkey;
```

Comment:
This appears in the original version too - @moneromooo-monero any clue why? This would appear to be unintentional (in the original code) after a quick scan of the current version of `wallet2.cpp`.

NOTE: This same line is also below in the update.

Reply:
Yes I am asking about this assignment, and whether it should be done at all. And yes, I know it is used several times in `wallet2`, thats why I asked about this assignment.